### DISCUSSION TITLE
How to debug the code?

### DISCUSSION THREAD

**Discussion by musabgultekin** [gildas-lormeau/SingleFile#1320] [Q&A] [answered]
These are my steps:

Clone the repo
Install npm packages
Make necessary changes
Ran build-extension.sh file.
Enable developer mode in chrome://extensions page.
Then drag and drop exported extension zip file to the chrome extensions page.

But I'm not sure exactly. Can you let me know which steps to take.

I'm using  MacOS Monterey (12.0.1), with Google Chrome 96.0.4664.55

**Comment by gildas-lormeau** [comment, marked as answer]
You don't need to generate the zip file actually. Just run `npx rollup -c rollup.config.dev.js` to build the code for debugging purposes, and install the extension in developer mode from the folder where you cloned the project.

**Comment by musabgultekin** [comment]
Thank you!

**Comment by gildas-lormeau** [comment]
You're welcome :)

**Comment by theowenyoung** [comment]
@gildas-lormeau Hi, I tried to run `npm run dev`, and it works, but how can I run it in a watch mode?

should I run this command `npm run dev` every time when I make some changes?

**Comment by gildas-lormeau** [comment]
@theowenyoung You have to run the command **and** refresh the extension in Chrome. That's why I run it manually on my end.

**Comment by grematt** [comment]
@gildas-lormeau Hey, when I run ```npx rollup -c rollup.config.js``` I get told that single-file-core/single-file.js cannot be found. Clearly I am missing something (maybe to generate these files?) as I do not see them present anywhere in the project. Any idea what I'm doing wrong? Thanks.

**Reply by gildas-lormeau** [reply]
Did you run `npm install`?

**Reply by grematt** [reply]
@gildas-lormeau yes, and I’ve tried to run it using npm run dev too but of course that gets stuck at the same point. I assume the order is just npm install -> npm run dev and that’s its?

**Reply by gildas-lormeau** [reply]
If you want to use `npm run dev`, then you have to clone https://github.com/gildas-lormeau/single-file-core and make sure it's in the same parent folder as SingleFile, e.g. `project-single-file/single-file-core` and `project-single-file/SingleFile`.

**Reply by grematt** [reply]
So if I just what to use npx rollup -c rollup.config.dev.js I don’t have to do that? Just npm install then said command?

**Reply by gildas-lormeau** [reply]
`npm run dev` is a shortcut for `npx rollup -c rollup.config.dev.js`. In both cases, you need to clone the source code of `single-file-core`.

**Reply by grematt** [reply]
Thanks, it is working now.
